### DISCUSSION TITLE
Your raw query had an incorrect number of parameters. Expected: `0`, actual: `2`. Prisma Prepared Statements

### DISCUSSION THREAD

**Discussion by Shamir-j** [prisma/prisma#13439] [Q&A] [answered]
Please go soft on me this is my first time asking here questions. 
I am having a scenario that is causing my query to fail when I try to template it with variables. But when I pass in the static values it works.
The first image shows a query with static values 

![Screenshot from 2022-05-22 00-48-27](https://user-images.githubusercontent.com/66664840/169670006-0c99c62b-a2f2-4e0e-9be3-f4cc014342ee.png)

the second image is the response of the above logic.
![Screenshot from 2022-05-22 00-49-50](https://user-images.githubusercontent.com/66664840/169670025-cec6696a-58b4-4c4b-afc7-6d173f776d23.png)

Now observe what happens when I change the query to use the templated prepared statements.
![Screenshot from 2022-05-22 00-53-13](https://user-images.githubusercontent.com/66664840/169670126-c5b96e2f-3365-4bee-b89c-fa99fda2bc72.png)

The response I get

![Screenshot from 2022-05-22 00-53-56](https://user-images.githubusercontent.com/66664840/169670150-75f5b33d-1ee7-4a45-92ff-dfcfc3ba70d7.png)

**Comment by nurul3101** [comment, marked as answer]
Hey @Shamir-j 👋 
Can you try if you get the same error by enabling the `improvedQueryRaw` preview feature?
Here's a reference on how you could enable it: [improvedQueryRaw](https://github.com/prisma/prisma/releases/tag/3.14.0)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
